### PR TITLE
Added feature flag for the new profile

### PR DESCRIPTION
**Pull request by acelaya**
This PR goes together with https://github.com/hypothesis/h/pull/7817, introducing a new feature flag which will allow the client to dynamically display an entry point for the new profile if desired.

### CODE REVIEW

File: h/models/feature.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "html_side_by_side": "Enable side-by-side mode for web pages in the client",
     "pdf_custom_text_layer": "Use custom text layer in PDFs for improved text selection",
     "styled_highlight_clusters": "Style different clusters of highlights in the client",
+    "client_new_profile": "Display entry point to new profile in the client's user context menu",
```
Comment (lyzadanger) [inline h/models/feature.py:18]: I think the distinction this feature flag makes is related to how we provide user preferences. That it is available in the user menu is a little less essential to the feature. Perhaps we could use

```suggestion
    "client_user_profile": "Enable client-side user profile and preferences management",
```

That gives us more flexibility: the point being that when the flag is enabled, the client is responsible for whatever is needed for profile and preferences management, however it sees fit. 

My usual regrets for being so pedantic...

**Review by lyzadanger — COMMENTED** [review]
Pedantry warning! I have a couple thoughts about the naming of this flag.

File: h/models/feature.py (hunk 1 above)
Comment (acelaya) [reply h/models/feature.py:18]: Absolutely agree. I admire how you always find the right words to describe something 😄

**Review by lyzadanger — APPROVED** [review]
Thanks!

**Pull request merged**